File: app/src/main/java/com/example/pshiksha/login/SplashScreenActivity.java
package com.example.pshiksha.login;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.example.pshiksha.databinding.ActivitySplashScreenBinding;
import com.example.pshiksha.login.LoginActivity;
import com.example.pshiksha.services.ServicesActivity;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import java.util.Objects;

public class SplashScreenActivity extends AppCompatActivity {

    ActivitySplashScreenBinding binding;
    FirebaseAuth auth;
    FirebaseUser user;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySplashScreenBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        auth = FirebaseAuth.getInstance();
        user = auth.getCurrentUser();

        Objects.requireNonNull(getSupportActionBar()).hide();

        new Handler().postDelayed(() -> {
            Intent intent;
            if (user != null) {
                intent = new Intent(getApplicationContext(), ServicesActivity.class);
            } else {
                intent = new Intent(getApplicationContext(), LoginActivity.class);
            }
            startActivity(intent);
            finish();
        }, 2000);
    }
}

File: app/src/main/java/com/example/pshiksha/login/SplashScreenActivity.kt
package com.example.pshiksha.login

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.animation.AnimationUtils
import androidx.appcompat.app.AppCompatActivity
import com.example.pshiksha.databinding.ActivitySplashScreenBinding
import com.example.pshiksha.main.MainActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser

class SplashScreenActivity : AppCompatActivity() {
    private lateinit var binding: ActivitySplashScreenBinding
    var auth: FirebaseAuth? = null
    var user: FirebaseUser? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySplashScreenBinding.inflate(layoutInflater)

        setContentView(binding.root)
        supportActionBar?.hide()

        auth = FirebaseAuth.getInstance()
        user = auth!!.currentUser

        binding.linearLayout.animation =
            AnimationUtils.loadAnimation(applicationContext, android.R.anim.fade_in)

        Handler(Looper.getMainLooper()).postDelayed({
            if (user != null) {
                startActivity(
                    Intent(
                        applicationContext,
                        MainActivity::class.java
                    )
                )
            } else {
                startActivity(Intent(applicationContext, LoginActivity::class.java))
            }
            finishAffinity()
        }, 1000)
    }
}